src/solver_nonlin_nox/4C_solver_nonlin_nox_vector.hpp

```diff
@@ -19,36 +22,99 @@ namespace NOX
       Teuchos::RCP<::NOX::Abstract::Vector> clone(
           ::NOX::CopyType type = ::NOX::DeepCopy) const override;
+
+      //! Get reference to underlying Epetra vector.
+      Core::LinAlg::Vector<double>& get_linalg_vector();
+
+      //! Get const reference to underlying Epetra vector.
+      const Core::LinAlg::Vector<double>& get_linalg_vector() const;
+
+      //! Initialize every element of this vector with @p gamma .
+      ::NOX::Abstract::Vector& init(double gamma) override;
+
+      //! Initialize each element of this vector with a random value.
+      ::NOX::Abstract::Vector& random(bool, int) override;
+
+      //! Put element-wise absolute values of source vector @p y into this vector.
+      ::NOX::Abstract::Vector& abs(const ::NOX::Abstract::Vector& y) override;
+
+      //! Put element-wise reciprocal of source vector @p y into this vector.
+      ::NOX::Abstract::Vector& reciprocal(const ::NOX::Abstract::Vector& y) override;
+
+      //! Scale each element of this vector by @p gamma .
+      ::NOX::Abstract::Vector& scale(double gamma) override;
+
+      //! Scale this vector element-by-element by the vector @p a .
+      ::NOX::Abstract::Vector& scale(const ::NOX::Abstract::Vector& a) override;
+
+      //! Compute x = (alpha * a) + (gamma * x) where x is this vector.
+      ::NOX::Abstract::Vector& update(
+          double alpha, const ::NOX::Abstract::Vector& a, double gamma = 0.0) override;
+
+      //! Compute x = (alpha * a) + (beta * b) + (gamma * x) where x is this vector.
+      ::NOX::Abstract::Vector& update(double alpha, const ::NOX::Abstract::Vector& a, double beta,
+          const ::NOX::Abstract::Vector& b, double gamma = 0.0) override;
+
+      //! Return the vector norm.
+      double norm(
+          ::NOX::Abstract::Vector::NormType type = ::NOX::Abstract::Vector::TwoNorm) const override;
+
+      //! Return the vector's weighted 2-norm.
+      double norm(const ::NOX::Abstract::Vector&) const override;
+
+      //! Return the inner product of the vector with @p y.
+      double innerProduct(const ::NOX::Abstract::Vector& y) const override;
+
+      //! Return the length of vector.
+      ::NOX::size_type length() const override;
+
+      //! Get reference to underlying Epetra vector - temporary and is to be removed
+      Epetra_Vector& getEpetraVector();  // NOLINT(readability-identifier-naming)
+
+      //! Get const reference to underlying Epetra vector - temporary and is to be removed
+      const Epetra_Vector& getEpetraVector() const;  // NOLINT(readability-identifier-naming)
+
+      // Create a view-like NOX::Epetra::Vector wrapper - temporary and is to be removed
+      operator ::NOX::Epetra::Vector();
+
+      // Create a view-like NOX::Epetra::Vector wrapper - temporary and is to be removed
+      operator ::NOX::Epetra::Vector() const;
+
+     protected:
+      //! Pointer to a storage vector owned by this object
+      std::shared_ptr<Core::LinAlg::Vector<double>> linalg_vec;
```

Comment:
We have `OwnerOrView` which we use a member for linalg wrappers. This enables most use cases of converting to/from `LinAlg::Vector`.

Reply:
Yes, indeed. But I do not see significant benefits from using it in this context. Actually here I may see just the opposite (again from my experience when debugging these changes): the ownership scheme in 4C gets out of control so easy. Semantically, `linalg_vec` should not be a view: it has to be a shared object indeed, ensuring its lifetime is essential for functionality of `NOX::Nln::Vector`. Maybe I do not fully understand the use cases of `OwnerOrView`.

Reply:
I guess you are right that this class will become more of a non-owning adapter to make "our" vector (which happens to be a Trilinos vector under the hood) compatible with NOX :+1: Similar to how you did it in deal.II